### DISCUSSION TITLE
Foundry advanced - ERC20 - Unable to compile with incompatible solidity versions.

### DISCUSSION THREAD

**Discussion by helloram007** [Cyfrin/foundry-full-course-cu#1905] [General]
1.  created a new forge init project
2.  create a token and deployment script. ran build was succesfful.
3.  copied makefile from earlier project and removed unneccessary steps sa menitoend in the course.
4.  when i ran ``` make deploy```, it had deleted all the libraries under lib/
5.  tried to manually install forge-std lib, but running "make deploy" fails.
6.  tried adding manually remppaings to forge-std, but no luck.

here is the error.

> [⠃] Compiling...2024-06-29T08:26:28.601922Z ERROR foundry_compilers::resolver: failed to resolve versions
[⠊] Compiling...
Error: 
Found incompatible Solidity versions:
script/DeployOurToken.sol =0.8.18 imports:
    lib/forge-std/src/Script.sol >=0.6.2, <0.9.0
    src/OurToken.sol =0.8.26
    lib/forge-std/src/console.sol >=0.4.22, <0.9.0
    lib/forge-std/src/console2.sol >=0.4.22, <0.9.0
    lib/forge-std/src/safeconsole.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdChains.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdCheats.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdJson.sol >=0.6.0, <0.9.0
    lib/forge-std/src/StdMath.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdStorage.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdStyle.sol >=0.4.22, <0.9.0
    lib/forge-std/src/StdUtils.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Base.sol >=0.6.2, <0.9.0
    lib/openzeppelin-contracts/contracts/token/ERC20/ERC20.sol ^0.8.20
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdStorage.sol >=0.6.2, <0.9.0
    lib/forge-std/src/console2.sol >=0.4.22, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/interfaces/IMulticall3.sol >=0.6.2, <0.9.0
    lib/forge-std/src/mocks/MockERC20.sol >=0.6.2, <0.9.0
    lib/forge-std/src/mocks/MockERC721.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/forge-std/src/StdStorage.sol >=0.6.2, <0.9.0
    lib/forge-std/src/Vm.sol >=0.6.2, <0.9.0
    lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Metadata.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/utils/Context.sol ^0.8.20
    lib/openzeppelin-contracts/contracts/interfaces/draft-IERC6093.sol ^0.8.20
    lib/forge-std/src/interfaces/IERC20.sol >=0.6.2
    lib/forge-std/src/interfaces/IERC721.sol >=0.6.2
    lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol ^0.8.20
    lib/forge-std/src/interfaces/IERC165.sol >=0.6.2

**Comment by EngrPips** [comment]
Hello @helloram007, The way I have always solved this issue is  to define a specific solidity compiler version in the `foundry.toml` file and enforce any contract or library to object version to use it by specifically changing their solidity version to what i have set. You can try this and let's if it resolve your issue.

**Reply by helloram007** [reply]
Thanks @EngrPips. i was able to run the tests now.
THis is my finding.
1) i added solc_version 0.8.26 and ran the tests, it went well.but there were few contracts with 0.8.0 versions, so i have to adjust it accordingly the version.
2) ran with commenting solc-version, it ran well again. i have adjusted solc to latest, so contract version i am using ^0.8.0 so it was picking and running tests fine.

dont know if this is my understanding.

But anyway, you provided the way.
Thanks Once again/

**Reply by EngrPips** [reply]
You are always welcome, Friend. I am glad you unblocked yourself.
